components/script/dom/document.rs

```diff
@@ -1575,6 +1612,97 @@ impl Document {
         ScriptThread::mark_document_with_no_blocked_loads(self);
     }
 
+    // https://html.spec.whatwg.org/multipage/#unloading-documents
```

Comment:
nit: https://html.spec.whatwg.org/multipage/#prompt-to-unload-a-document
Label sub-steps